<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>llm_tool_editor</class>
 <widget class="QDialog" name="llm_tool_editor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>LLM Tool Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="title_label">
     <property name="text">
      <string>Tool Editor</string>
     </property>
     <property name="styleSheet">
      <string>font-size: 16px; font-weight: bold;</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="name_label">
       <property name="text">
        <string>Tool Name:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="name_input">
       <property name="placeholderText">
        <string>e.g., calculate_fibonacci</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="display_name_label">
       <property name="text">
        <string>Display Name:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="display_name_input">
       <property name="placeholderText">
        <string>e.g., Calculate Fibonacci</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="description_label">
       <property name="text">
        <string>Description:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QTextEdit" name="description_input">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>80</height>
        </size>
       </property>
       <property name="placeholderText">
        <string>What does this tool do? Be clear and concise.</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="code_label">
       <property name="text">
        <string>Python Code:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QTextEdit" name="code_input">
       <property name="font">
        <font>
         <family>Monospace</family>
        </font>
       </property>
       <property name="placeholderText">
        <string>@tool
def my_tool(param: str) -> str:
    """Tool description."""
    # Your code here
    return result</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QCheckBox" name="enabled_checkbox">
       <property name="text">
        <string>Enabled</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="info_layout">
     <item>
      <widget class="QLabel" name="safety_label">
       <property name="text">
        <string>⚠️ Safety Check: Not Run</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="validate_button">
       <property name="text">
        <string>Validate Code</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Save</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>llm_tool_editor</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>llm_tool_editor</receiver>
   <slot>reject()</slot>
  </connection>
  <connection>
   <sender>validate_button</sender>
   <signal>clicked()</signal>
   <receiver>llm_tool_editor</receiver>
   <slot>on_validate_clicked()</slot>
  </connection>
 </connections>
 <slots>
  <slot>on_validate_clicked()</slot>
 </slots>
</ui>
